index/scorch/persister.go

```diff
@@ -268,13 +279,20 @@ func (s *Scorch) parsePersisterOptions() (*persisterOptions, error) {
 	return &po, nil
 }
 
-func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot) error {
-	persisted, err := s.persistSnapshotMaybeMerge(snapshot)
-	if err != nil {
-		return err
-	}
-	if persisted {
-		return nil
+func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot,
+	po *persisterOptions) error {
+	// With ActiveMergeMemorySegments disabled, perform in-memory segment
+	// merging only when there is no memory pressure, else persister
+	// always performs the memory segments merge before persistence.
+	if po.ActiveMergeMemorySegments ||
+		(!po.ActiveMergeMemorySegments && s.paused() == 0) {
```

Comment:
This can be simplified to:
`if po.ActiveMergeMemorySegments || s.paused() == 0 {`

Reply:
(whoops -- hit the wrong button.  unresolving Abhinav's comment!)

Reply:
One other thought (and not sure how this goes in this situation), but one thing I like to see in cases like this is whether the new option can represent a range of values.  Like...

if s.paused() >= SomeNewOptionGoodLuckNamingMe {
    ....
}

The default for that SomeNewOptionGoodLuckNamingMe might be 0, for example, which means to super aggressively in-memory merge during persistence.  I might have reversed the >= versus <=, but if that way one single advanced config option can cover more ground.